### PR TITLE
Feature new sdk implement download events with basic pagination

### LINKED ISSUE #261: Implement download events with basic pagination and limit

### PR DESCRIPTION
**Pull request by xurxodev**
close #261

### CODE REVIEW

**Comment by xurxodev** [comment]
@ifoche I have realized optimizations using Gradle tasks

File: core/src/androidTest/java/org/hisp/dhis/android/core/event/EventCallMockIntegrationShould.java [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+
+        EventStoreImpl eventStore = new EventStoreImpl(databaseAdapter());
+
+        List<Event> downloadedEvents = eventStore.querySingleEvents();
+
+        assertThat(downloadedEvents.size(), is(pageLimit));
+    }
+
+    @Test
+    public void remove_data_values_removed_in_server_after_second_events_download()
```
Comment (ifoche) [inline core/src/androidTest/java/org/hisp/dhis/android/core/event/EventCallMockIntegrationShould.java]: @xurxodev this test is failing. Can you give a look to it?

File: core/src/androidTest/java/org/hisp/dhis/android/core/event/EventCallMockIntegrationShould.java [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+    @Override
+    @After
+    public void tearDown() throws IOException {
+        super.tearDown();
+
+        dhis2MockServer.shutdown();
+    }
+
+    @Test
+    public void download_events_according_to_default_query() throws Exception {
```
Comment (ifoche) [inline core/src/androidTest/java/org/hisp/dhis/android/core/event/EventCallMockIntegrationShould.java]: @xurxodev this test is failing. Can you give a look to it?

File: core/src/test/java/org/hisp/dhis/android/core/event/EventCallShould.java [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void throw_illegal_argument_exception_if_uids_size_exceeds_the_limit() {
+        EventCall eventCall = givenAEventCallByUIds(MAX_UIDS + 1);
+    }
+
+    @Test
+    public void create_event_call_if_uIds_size_does_not_exceeds_the_limit() {
+        EventCall eventCall = givenAEventCallByUIds(MAX_UIDS);
```
Comment (ifoche) [inline core/src/test/java/org/hisp/dhis/android/core/event/EventCallShould.java]: @xurxodev don't we miss an assertion here? It looks like we're not checking the eventCall is created, right?

**Comment by xurxodev** [comment]
@ifoche I have added assertion to create_event_call_if_uIds_size_does_not_exceeds_the_limit .
All test is passed for me, also remove_data_values_removed_in_server_after_second_events_download and download_events_according_to_default_query. 
Please can you review that you are using in-memory database (databaseName = null in AbsStoreTestCase)? Disk database not working with the current approach of all tests.

**Comment by ifoche** [comment]
@xurxodev thanks for the assertion. I've reviewed and I'm using in memory db indeed...so I'm not sure why it doesn't work for me...maybe we can give a look together later today?

**Comment by ifoche** [comment]
@xurxodev just sent you the files that are breaking the assertion for one of the tests. It looks like dates are the reason for that problem in the assertion. Dates have a shift of a couple of hours...so strictly they're not the same objects.

**Pull request merged**